sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/OptimizeSkewedJoin.scala

```diff
@@ -250,15 +253,28 @@ object OptimizeSkewedJoin extends AQEShuffleReadRule {
       // SHJ
       //   Shuffle
       //   Shuffle
-      optimizeSkewJoin(plan)
+      val optimized =
+        EnsureRequirements(requiredDistribution.isDefined, requiredDistribution)
```

Comment:
shall we simply pass in the `EnsureRequirements` instance to `OptimizeSkewedJoin`?

Reply:
seems better, `OptimizeSkewedJoin` doesn't need to care `requiredDistribution`